src/services/CleanupService.ts

```diff
@@ -404,110 +406,218 @@ export const clearLocalstorageForThisDomain = async (
-            },
-            debug,
-          );
-          throw e;
-        });
-      return true;
+  cadLog(
+    {
+      msg: `CleanupService.clearSiteDataForThisDomain: Received ${siteData} clean request for ${hostname}.`,
+    },
+    debug,
+  );
+  const domains = prepareCleanupDomains(hostname, isChrome(state.cache));
+  if (siteData === 'All') {
+    for (const sd of [
+      'cache',
+      'cookies',
+      'indexedDB',
+      'localStorage',
+      'pluginData',
+      'serviceWorkers',
+    ]) {
+      await removeSiteData(sd, isChrome(state.cache), domains, debug);
     }
-    if (isChrome(state.cache)) {
-      const cleanList: string[] = [];
-      for (const domain of domains) {
-        for (const d of prepareCleanupDomains(domain)) {
-          if (d !== '' && d.trim() !== '') {
-            cleanList.push(`http://${d}`);
-            cleanList.push(`https://${d}`);
-          }
-        }
-      }
-      if (cleanList.length === 0) return false;
-      const origins = [...new Set(cleanList)];
-      cadLog(
-        {
-          msg: `CleanupService.otherBrowsingDataCleanup: Origins sent to Chrome LocalStorage Cleanup`,
-          x: origins,
-        },
+  } else {
+    await removeSiteData(
+      `${siteData[0].toLowerCase()}${siteData.slice(1)}`,
+      isChrome(state.cache),
+      domains,
+      debug,
+    );
+  }
+  return true;
+};
+
+export const removeSiteData = async (
+  siteData: string,
+  isChrome: boolean,
+  domains: string[],
+  debug: boolean,
+): Promise<boolean> => {
+  const listName = isChrome ? 'origins' : 'hostnames';
+  cadLog(
+    {
+      msg: `CleanupService.removeSiteData: Cleanup of ${listName} for ${siteData}:`,
+      x: domains,
+    },
+    debug,
+  );
+  try {
+    await browser.browsingData.remove(
+      {
+        [listName]: domains,
+      },
+      {
+        [siteData]: true,
+      },
+    );
+    return true;
+  } catch (e) {
+    cadLog(
+      {
+        msg: `CleanupService.removeSiteData:  browser.browsingData.remove of ${listName} for ${siteData} returned an error:`,
+        type: 'error',
+        x: e,
+      },
+      debug,
+    );
+    throwErrorNotification(e);
+    return false;
+  }
+};
+
+/** This will use the browsingData's hostname/origin attribute to delete any extra browsing data */
+export const otherBrowsingDataCleanup = async (
+  state: State,
+  isSafeToCleanObjects: CleanReasonObject[],
+): Promise<ActivityLog['browsingDataCleanup']> => {
+  const chrome = isChrome(state.cache);
+  const debug = getSetting(state, 'debugMode') as boolean;
+  const browsingDataResult: ActivityLog['browsingDataCleanup'] = {};
+  if (
+    getSetting(state, 'cacheCleanup') &&
+    ((isFirefoxNotAndroid(state.cache) && state.cache.browserVersion >= '78') ||
+      chrome)
+  ) {
+    browsingDataResult.cache = await cleanSiteData(
+      'Cache',
+      isSafeToCleanObjects,
+      chrome,
+      debug,
+    );
+  }
+  if (
+    getSetting(state, 'indexedDBCleanup') &&
```

Comment:
Don't have to do it in this PR but just looking at this, I think `indexedDBCleanup` and others for getSetting should be part of an enum string. Otherwise, it would be prone to misspellings and refactoring nightmares (like changing the name).

Just something to note for the future.